src/main/java/org/jabref/logic/bibtex/DuplicateCheck.java

```diff
@@ -98,7 +102,16 @@ public static boolean isDuplicate(BibEntry one, BibEntry two, BibDatabaseMode bi
         return req[0] >= DuplicateCheck.duplicateThreshold;
     }
 
-    private static boolean hasSameIdentifier(BibEntry one, BibEntry two) {
+    private static boolean haveDifferentEditions(BibEntry one, BibEntry two) {
+        if (one.getField(FieldName.EDITION).isPresent() && two.getField(FieldName.EDITION).isPresent()) {
+            if (!one.getField(FieldName.EDITION).get().equals(two.getField(FieldName.EDITION).get())) {
```

Comment:
Yeah, I'm aware of that. That's why I first checked both to be present, and then called `.get()` on both to explicitly state that this line will only work if there is a value in the optional.